pkg/core/diagnostics.go

```diff
@@ -0,0 +1,69 @@
+package core
+
+import "github.com/cloudquery/cq-provider-sdk/provider/diag"
+
+type FetchDiagnostic struct {
+	base diag.Diagnostic
```

Comment:
no need for base, just embed it directly. That way you don't need the overloaded methods unless you really want to overload.